### PR TITLE
docs(README): add awesome-go badge to README

### LINKED ISSUE #866: Include awesome-go badge in README
**Describe the feature or problem you'd like to solve**

I would like to bring visibility to the fact the glab repo is now listed in the https://github.com/avelino/awesome-go project.

**Propose a Solution**

The awesome-go repo provides a dedicated badge listed repos can include.
https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md#congrats-your-project-got-accepted---what-now

**Additional context**

glab was only recently added to the awesome-go list in this [PR](https://github.com/avelino/awesome-go/pull/3805)

### PR DESCRIPTION
**Pull request by TheArchDev**
glab is now listed in awesome-go repo so follow guidance to add badge
https://github.com/avelino/awesome-go/blob/master/CONTRIBUTING.md

Delivers #866

## Description

The awesome-go has a list of Go-based tools around version control.
https://github.com/avelino/awesome-go#version-control
glab has recently been added to this list.
A repo may indicate its presence on that list through a dedicated badge.

## Related Issue

Resolves #866 

## Screenshots (if appropriate)
Screenshot of README in Github on feature branch.
Screenshot 2021-10-07 at 11 35 08

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)
- [x] Documentation
- [ ] Chore (Related to CI or Packaging to platforms)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -6,6 +6,8 @@
 [![Gitter](https://badges.gitter.im/glabcli/community.svg)](https://gitter.im/glabcli/community?utm_source=badge&utm_medium=badge&utm_campaign=pr-badge)
 [![Reddit](https://img.shields.io/reddit/subreddit-subscribers/glab_cli?style=social)](https://reddit.com/r/glab_cli)
 [![Twitter Follow](https://img.shields.io/twitter/follow/glab_cli?style=social)](https://twitter.com/glab_cli)
+[![Mentioned in Awesome Go](https://awesome.re/mentioned-badge.svg)](https://github.com/avelino/awesome-go)
+
```
Comment (profclems) [inline README.md]: nit: the extra line break is not needed

```suggestion
```

**Review by profclems — APPROVED** [review]
Thanks for this @TheArchDev!

File: README.md [hunk 2]
```diff
@@ -8,7 +8,6 @@
 [![Twitter Follow](https://img.shields.io/twitter/follow/glab_cli?style=social)](https://twitter.com/glab_cli)
 [![Mentioned in Awesome Go](https://awesome.re/mentioned-badge.svg)](https://github.com/avelino/awesome-go)
```
Comment (profclems) [inline README.md]: Maybe we could use more specific URL: `https://github.com/avelino/awesome-go#version-control`

**Review by profclems — APPROVED** [review]

File: README.md (hunk 2 above)
Comment (TheArchDev) [reply README.md]: Great idea! I have added another commit to do this. Thanks for the help @profclems

**Pull request merged**